src/api/firebase.js

```diff
@@ -71,15 +78,30 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 		dateNextPurchased: getFutureDate(daysUntilNextPurchase),
 		name: itemName,
 		totalPurchases: 0,
+		checked: false,
 	});
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+/**
+ * update a user's list item in Firebase
+ * @param {*} listId The list token we are modifying its data.
+ * @param {*} itemId The item id.
+ * @param {*} checked A boolean value indicating whether the item is purchased.
+ */
+export async function updateItem(listId, itemId, checked) {
+	const itemRef = doc(db, listId, itemId);
+	const itemDoc = await getDoc(itemRef);
+	const { dateLastPurchased, totalPurchases } = itemDoc.data();
+
```

Comment:
I love how you are using destructuring to extract the necessary fields from the document!